crates/legacy/src/service.rs

```diff
@@ -1044,13 +1082,34 @@ impl<Types: NodeType> AcceptsTxnSubmits<Types> for ProxyGlobalState<Types> {
             txns.len(),
             txns.iter().map(|txn| txn.commit()).collect::<Vec<_>>()
         );
+        for txn in txns.clone() {
+            let _ = self
+                .global_state
+                .write_arc()
+                .await
+                .set_tx_status(txn.commit(), TransactionStatus::Pending);
+        }
         let response = self
             .global_state
             .read_arc()
             .await
-            .submit_client_txns(txns)
+            .submit_client_txns(txns.clone())
             .await;
 
+        let pairs: Vec<(<Types as NodeType>::Transaction, Result<_, _>)> = (0..txns.len())
+            .map(|i| (txns[i].clone(), response[i].clone()))
+            .collect();
+        for (txn, res) in pairs {
+            if let Err(some) = res {
```

Comment:
Why not set transaction status to Pending in an `else` branch here?

Reply:
The initialization status of all the transaction will be `Pending`, therefore we only set `sequenced` or `rejected` additionally.

Reply:
Yeah, what I mean is that we could cut down on number of `set_txn_status` calls by skipping the first loop and setting transactions to `Pending` only when `submit_client_txns` returned `Ok`. I don't think a client would like to know the status of their transaction _before_ their call to submit them has finished in the first place.

Reply:
Aha...that's a good point.